### PR TITLE
Validator unit tests.

### PR DESCRIPTION
**Pull request by ntieman**

### CODE REVIEW

**Comment by john9hoff** [comment]
This is great, you're on the right track!

There's too many lines of code here and I think you can easily cut it down by making use of some kind of a loop. For example, you could make an array of "bad values" for endpoints, e.g., [na, fe, eu], then loop through it and expect all of those to throw errors.

How you want to go about this is totally up to you, though!

File: test/util/validator.test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+const { validateInputJson } = require('../../src/util/validator')
+
+test('validateInputJson', () => {
+  const goodRequest = {
```
Comment (john9hoff) [inline test/util/validator.test.js]: Rename this to `request`, and make it into its own test that passes successful validation.

File: test/util/validator.test.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  const stringFields = [ 'currencyCode', 'partnerId', 'accessKey', 'secretKey' ]
+  const enumFields = [ 'environment', 'endpoint' ]
+  const badStrings = [ 35, [], '' ]
+  const goodEnums = {
+    environment: [ 'sandbox', 'production' ],
+    endpoint: [ 'NA', 'EU', 'FE' ]
+  }
+
+  expect(() => validateInputJson(goodRequest)).not.toThrowError()
+  expect(() => validateInputJson(null)).toThrowError('JSON cannot be null')
```
Comment (john9hoff) [inline test/util/validator.test.js]: Make this it's own test. E.g., test('Null JSON throws error', () => { //code here }

File: test/util/validator.test.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  expect(() => validateInputJson({ ...goodRequest, amount: '20' })).not.toThrowError()
+  expect(() => validateInputJson({ ...goodRequest, amount: 999999 })).not.toThrowError()
+  
+  expect(() => validateInputJson({ ...goodRequest, amount: 0 })).toThrowError()
```
Comment (john9hoff) [inline test/util/validator.test.js]: Make this into its own test. E.g., test('Invalid amount throws error', () => { //code here }

Make an array of invalid amounts and loop through them

File: test/util/validator.test.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  expect(() => validateInputJson({ ...goodRequest, amount: '0' })).toThrowError()
+  expect(() => validateInputJson({ ...goodRequest, amount: -20 })).toThrowError()
+  expect(() => validateInputJson({ ...goodRequest, amount: '-20' })).toThrowError()
+  expect(() => validateInputJson({ ...goodRequest, amount: '' })).toThrowError()
+
+  stringFields.forEach(field => {
```
Comment (john9hoff) [inline test/util/validator.test.js]: Make it's own test, e.g. test('Valid strings', () => { //code here }

Put the array of strings in this test

File: test/util/validator.test.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    expect(() => validateInputJson({ ...goodRequest, [field]: '35' })).not.toThrowError()
+
+    badStrings.forEach(badString => expect(() => validateInputJson({ ...goodRequest, [field]: badString })).toThrowError())
```
Comment (john9hoff) [inline test/util/validator.test.js]: Make it its own test called 'Non-string values throws error'

declare the badStrings array here

File: test/util/validator.test.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  })
+
+  enumFields.forEach(field => {
```
Comment (john9hoff) [inline test/util/validator.test.js]: Make its own test

File: test/util/validator.test.js [hunk 7]
```diff
@@ -12,43 +12,59 @@ test('validateInputJson', () => {
   }
 
   const stringFields = [ 'currencyCode', 'partnerId', 'accessKey', 'secretKey' ]
-  const enumFields = [ 'environment', 'endpoint' ]
-  const badStrings = [ 35, [], '' ]
-  const goodEnums = {
+  
+  const enums = {
```
Comment (john9hoff) [inline test/util/validator.test.js:16]: Move to valid enums test

File: test/util/validator.test.js [hunk 8]
```diff
@@ -1,7 +1,7 @@
 const { validateInputJson } = require('../../src/util/validator')
 
-test('validateInputJson', () => {
-  const goodRequest = {
+describe('validateInputJson', () => {
+  const request = {
```
Comment (john9hoff) [inline test/util/validator.test.js:4]: Move to 'Good request' test

File: test/util/validator.test.js [hunk 9]
```diff
@@ -12,43 +12,59 @@ test('validateInputJson', () => {
   }
 
   const stringFields = [ 'currencyCode', 'partnerId', 'accessKey', 'secretKey' ]
```
Comment (john9hoff) [inline test/util/validator.test.js:14]: Move to strings test

File: test/util/validator.test.js (hunk 8 above)
Comment (ntieman) [reply test/util/validator.test.js]: The enum object is used by multiple tests.

File: test/util/validator.test.js (hunk 7 above)
Comment (ntieman) [reply test/util/validator.test.js:16]: The enum object is used by multiple tests.

File: test/util/validator.test.js (hunk 8 above)
Comment (ntieman) [reply test/util/validator.test.js:4]: The request object is used by almost all tests.

File: test/util/validator.test.js (hunk 9 above)
Comment (ntieman) [reply test/util/validator.test.js:14]: The stringFields array is used by multiple tests.

**Pull request merged**